### PR TITLE
Allow to parse value literals

### PR DESCRIPTION
**Pull request by sungam3r**
I need to parse values returned from introspection request, mostly default values of arguments. Our current design allows to parse only entire GraphQL document. 

Related project - https://github.com/sungam3r/graphql-introspection-model

### CODE REVIEW

**Comment by sungam3r** [comment]
@Shane32 I'm rewriting https://github.com/sungam3r/graphql-introspection-model now. I decided to completely remove my own `SDLBuilder` class that prints schema and switch to `SDLPrinter` from parser project. To be able to do this I wrote `ASTConverter` that converts my introspection representations into AST representations of GraphQL-Parser. Doing this I noticed that I need to properly handle all literals returned as default values or arguments. Fortunately all code I need is already here, but not as public API.

**Review by Shane32 — APPROVED** [review]

File: src/GraphQLParser.ApiTests/GraphQLParser.approved.txt [hunk 1]
```diff
@@ -661,6 +661,7 @@ namespace GraphQLParser
     public static class Parser
     {
         public static GraphQLParser.AST.GraphQLDocument Parse(GraphQLParser.ROM source, GraphQLParser.ParserOptions options = default) { }
+        public static GraphQLParser.AST.GraphQLValue ParseValue(GraphQLParser.ROM source, GraphQLParser.ParserOptions options = default) { }
```
Comment (sungam3r) [inline src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: Just a note - it seems that we could make more methods public to allow parse concrete parts of document.

Comment (Shane32) [reply src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: Yeah I was thinking like "couldn't we just have Parse" return whatever was parsed?  Then cast it to whatever type was desired or something?  But the problem is that the parser wouldn't know the initial state -- if, for example, it was parsing a list of arguments or a list of directives.

We could perhaps add `Parse<T>` where T is a node type, and then a `switch typeof(T)` logic to bump into the correct code.

Comment (Shane32) [reply src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: Then you could write `Parse<GraphQLValue>` if you want that, or `Parse<List<GraphQLArgument>>` if you are parsing that, etc, with very little API changes and somewhat intuitive behavior.

Comment (Shane32) [reply src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: Not really a fan of a million different public `Parse...` methods.  Hopefully they are all exposed as protected methods or something so anyone CAN use them if needed.

Comment (sungam3r) [reply src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: This is a subject to discuss for v9.

Comment (sungam3r) [reply src/GraphQLParser.ApiTests/GraphQLParser.approved.txt:664]: #296

**Pull request merged**